### PR TITLE
Added Flag for Effects That Activate by Detaching Xyz Materials as cost.

### PR DESCRIPTION
**Pull request by TheOnePharaoh**

### CODE REVIEW

File: c74294676.lua [hunk 1]
```diff
@@ -56,10 +57,6 @@ end
 function c74294676.condition2(e,tp,eg,ep,ev,re,r,rp)
 	return Duel.GetCurrentChain()==0
 end
-function c74294676.cost2(e,tp,eg,ep,ev,re,r,rp,chk)
```
Comment (TheOnePharaoh) [inline c74294676.lua]: it's the same cost

File: c77631175.lua [hunk 2]
```diff
@@ -14,7 +14,7 @@ function c77631175.initial_effect(c)
 	--damage
 	local e2=Effect.CreateEffect(c)
 	e2:SetDescription(aux.Stringid(77631175,1))
-	e2:SetCategory(CATEGORY_DAMAGE)
+	e2:SetCategory(CATEGORY_ATKCHANGE+CATEGORY_DAMAGE)
```
Comment (TheOnePharaoh) [inline c77631175.lua]: Category was missing

**Comment by mercury233** [comment]
what is this flag for?

**Comment by purerosefallen** [comment]
why not a category but a property?

**Comment by TheOnePharaoh** [comment]
@purerosefallen it's a property because it's cost related not effect related.

**Comment by Hello23-Ygopro** [comment]
Thanks , thats a lot of scripts that got updated
but do you mind saying which cards need that flag ? Is it to help some unreleased ocg cards to function properly , or anime/manga/vg/etc, related , or maybe as a pre-caution for future effects ?

File: c77631175.lua (hunk 2 above)
Comment (Hello23-Ygopro) [inline c77631175.lua]: i dont think they care about missing Categorys anymore

**Pull request closed without merging**